src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js

```diff
@@ -63,218 +64,212 @@ const defaultProps = {
     preferredLocale: CONST.LOCALES.DEFAULT,
 };
 
-class BaseValidateCodeForm extends React.Component {
-    constructor(props) {
-        super(props);
-        this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
-        this.resendValidateCode = this.resendValidateCode.bind(this);
-        this.clearSignInData = this.clearSignInData.bind(this);
+function BaseValidateCodeForm(props) {
+    const [formError, setFormError] = useState({});
+    const [validateCode, setValidateCode] = useState(props.credentials.validateCode || '');
+    const [twoFactorAuthCode, setTwoFactorAuthCode] = useState('');
+    const [linkSent, setLinkSent] = useState(false);
 
-        this.state = {
-            formError: {},
-            validateCode: props.credentials.validateCode || '',
-            twoFactorAuthCode: '',
-            linkSent: false,
-        };
-    }
+    const wasVisible = usePrevious(props.isVisible);
+    const required2FA = usePrevious(props.account.requiresTwoFactorAuth);
+    const hadValidateCode = usePrevious(props.credentials.validateAndSubmitForm);
 
-    componentDidMount() {
-        if (!canFocusInputOnScreenFocus() || !this.inputValidateCode || !this.props.isVisible) {
+    const refValidateCode = useRef(null);
+    const inputValidateCode = refValidateCode.current;
+    const ref2FA = useRef(null);
+    const input2FA = ref2FA.current;
+
+    useEffect(() => {
+        if (!inputValidateCode || wasVisible || !props.isVisible || !canFocusInputOnScreenFocus()) {
             return;
         }
-        this.inputValidateCode.focus();
-    }
+        inputValidateCode.focus();
+    }, [inputValidateCode, props.isVisible, wasVisible]);
```

Comment:
Actually, I don't think `canFocusInputOnScreenFocus` can change in the lifetime of the app because it relies on user agent. So we might only need to check it in `componentDidMount`

Reply:
It won't change but the current logic used to relay on that function only on component mount, Now with the refactor it's treated as if that function was being used on componentDIdUpdate too. This is not necessary a bad change, just asked for a confirmation in the linked PR.